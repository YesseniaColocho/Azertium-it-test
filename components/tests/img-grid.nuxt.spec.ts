import { describe, it, expect} from 'vitest'
import { shallowMount } from '@vue/test-utils'

import imgGrid from '../img-grid.vue'

const imagesProp = [
  { id: 1 },
  { id: 2 },
  { id: 3 }
]

describe('image grid', () => {
    const component = () => shallowMount(
      imgGrid, 
      {
        props: {
          images: imagesProp
        }
      }
    )

    it('component renders image grid properly', () => {
      const wrapper = component()
      expect(wrapper).toBeTruthy()
    })

    it('component html renders properly', () => {
      const wrapper = component()
      expect(wrapper.html()).toMatchSnapshot()
    })
})