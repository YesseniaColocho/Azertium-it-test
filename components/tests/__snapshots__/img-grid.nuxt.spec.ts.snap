// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`image grid > component html renders properly 1`] = `
"<section data-v-0f1ff06a="" class="img-grid">
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
</section>"
`;

exports[`top bar > component html renders properly 1`] = `
"<section data-v-0f1ff06a="" class="img-grid">
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
  <img-square-stub data-v-0f1ff06a="" imageprop="[object Object]"></img-square-stub>
</section>"
`;
